kustomize/components/alloydb/README.md

```diff
@@ -0,0 +1,152 @@
+```
+
+## Grant the `cartservice`'s service account access to the AlloyDB database
+
+**Important note:** Your GKE cluster should have [Workload Identity enabled](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable).
+
+As a good practice, let's create a dedicated least privilege Google Service Account to allow the `cartservice` to communicate with the AlloyDB database and grab the database password from the Secret manager.:
+```bash
+gcloud iam service-accounts create ${ALLOYDB_USER_GSA_NAME} \
+    --display-name=${ALLOYDB_USER_GSA_NAME}
+
+gcloud projects add-iam-policy-binding ${PROJECT_ID} --member=serviceAccount:${ALLOYDB_USER_GSA_ID} --role=roles/alloydb.client
+gcloud projects add-iam-policy-binding ${PROJECT_ID} --member=serviceAccount:${ALLOYDB_USER_GSA_ID} --role=roles/secretmanager.secretAccessor
+
+gcloud iam service-accounts add-iam-policy-binding ${ALLOYDB_USER_GSA_ID} \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[default/${CARTSERVICE_KSA_NAME}]" \
+    --role roles/iam.workloadIdentityUser
+```
+
+## Deploy Online Boutique connected to an AlloyDB database
+
+To automate the deployment of Online Boutique integrated with AlloyDB you can leverage the following variation with [Kustomize](../..).
+
+From the `kustomize/` folder at the root level of this repository, execute these commands:
+```bash
+kustomize edit add component components/service-accounts
+kustomize edit add component components/alloydb
+```
+_Note: this Kustomize component will also remove the `redis-cart` `Deployment` and `Service` not used anymore._
+
+This will update the `kustomize/kustomization.yaml` file which could be similar to:
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- base
+components:
+- components/service-accounts
+- components/alloydb
+```
+
+Update current Kustomize manifest to target this Spanner database.
+```bash
+sed -i "s/PROJECT_ID_VAL/${PROJECT_ID}/g" components/alloydb/kustomization.yaml
+sed -i "s/ALLOYDB_PRIMARY_IP_VAL/${ALLOYDB_PRIMARY_IP}/g" components/alloydb/kustomization.yaml
+sed -i "s/ALLOYDB_USER_GSA_ID/${ALLOYDB_USER_GSA_ID}/g" components/alloydb/kustomize.yaml
+sed -i "s/ALLOYDB_DATABASE_NAME_VAL/${ALLOYDB_DATABASE_NAME}/g" components/alloydb/kustomization.yaml
+sed -i "s/ALLOYDB_TABLE_NAME_VAL/${ALLOYDB_TABLE_NAME}/g" components/alloydb/kustomization.yaml
+sed -i "s/ALLOYDB_SECRET_NAME_VAL/${ALLOYDB_SECRET_NAME}/g" components/alloydb/kustomization.yaml
+```
+
+You can locally render these manifests by running `kubectl kustomize .` as well as deploying them by running `kubectl apply -k .`.
+
+## Extra cleanup steps
+```bash
+gcloud compute addresses delete ${ALLOYDB_SERVICE_NAME} --global
+
+# Force takes care of cleaning up the instances inside the cluster automatically
+gcloud alloydb clusters delete ${ALLOYDB_CLUSTER_NAME} --force
+
+gcloud iam service-accounts delete ${ALLOYDB_USER_GSA_ID}
+
+gcloud secrets delete ${ALLOYDB_SECRET_NAME}
+```
```

Comment:
**Issue:**
See the [current rendering of this README.md here](https://github.com/GoogleCloudPlatform/microservices-demo/blob/d4d4d6ef603d293d02e3e719d31443dabe78dd98/kustomize/components/alloydb/README.md).
We'll need to tidy up some of the formatting.

Reply:
Cleaned up all the formatting (I think).